rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -2165,6 +2033,40 @@ rmw_create_client(
   DDSDataReader * response_datareader = nullptr;
   DDSReadCondition * read_condition = nullptr;
   ConnextDynamicClientInfo * client_info = nullptr;
+  std::string request_type_name;
+  std::string response_type_name;
+  const void * untyped_request_members;
+  const void * untyped_response_members;
+
+  DDS_DomainParticipantQos participant_qos;
+  DDS_ReturnCode_t status = participant->get_qos(participant_qos);
+  if (status != DDS_RETCODE_OK) {
+    RMW_SET_ERROR_MSG("failed to get participant qos");
+    return NULL;
+  }
+  untyped_request_members =
+    get_request_ptr(type_support->data, type_support->typesupport_identifier);
+  untyped_response_members = get_response_ptr(type_support->data,
+      type_support->typesupport_identifier);
+  request_type_name = _create_type_name(untyped_request_members, "srv",
+      type_support->typesupport_identifier);
+  response_type_name = _create_type_name(untyped_response_members, "srv",
+      type_support->typesupport_identifier);
+
+  request_type_code = _create_type_code(request_type_name, untyped_request_members,
+      type_support->typesupport_identifier);
+  if (!request_type_code) {
+    // error string was set within the function
+    goto fail;
+  }
```

Comment:
While this generic error condition works the error message will be less specific (just mentioning that some members are null without specifying which one).

Same below.